### ISSUE TITLE
`torch.overrides.get_overridable_functions()` throws a bunch of warnings

### ISSUE THREAD

**Issue by ngimel** [pytorch/pytorch#109842]
```
In [2]: out=torch.overrides.get_overridable_functions()
/root/.pyenv/versions/3.9.9/lib/python3.9/site-packages/torch/overrides.py:110: UserWarning: 'has_cuda' is deprecated, please use 'torch.backends.cuda.is_built()'
  torch.has_cuda,
/root/.pyenv/versions/3.9.9/lib/python3.9/site-packages/torch/overrides.py:111: UserWarning: 'has_cudnn' is deprecated, please use 'torch.backends.cudnn.is_available()'
  torch.has_cudnn,
/root/.pyenv/versions/3.9.9/lib/python3.9/site-packages/torch/overrides.py:117: UserWarning: 'has_mps' is deprecated, please use 'torch.backends.mps.is_built()'
  torch.has_mps,
/root/.pyenv/versions/3.9.9/lib/python3.9/site-packages/torch/overrides.py:118: UserWarning: 'has_mkldnn' is deprecated, please use 'torch.backends.mkldnn.is_available()'
  torch.has_mkldnn,
 ```
 which shouldn't be the case for torch's own functions called with correct arguments. 

cc @ezyang @gchanan @zou3519 @kadeng

**Comment by tringwald** [comment]
This also happens for other public functions in that module:

- `torch.overrides.get_ignored_functions()` 
- `torch.overrides.get_ignored_functions()`
- `torch.overrides.get_overridable_functions()`
- `torch.overrides.resolve_name()`
- `torch.overrides.is_tensor_method_or_property()`

I'm currently working on a PR to sidestep this problem.

**Comment by cpuhrsch** [comment]
Marking this high priority since there's also a PR that fixes it and we can discuss if this adds up to a wider problem.

**Comment by tringwald** [comment]
The patch wasn't included in `2.1.2`, but it is part of the upcoming `2.2` release.
